### PR TITLE
refactor: split dev requirements out of main requirements.txt

### PR DESCRIPTION
**Pull request by lovesegfault**
This is a re-try of #467 since it has come up as an issue again.

Hopefully this time I managed to fix the release workflow correctly.

### CODE REVIEW

**Comment by maffoo** [comment]
@95-martin-orion, what's the status of this? We would like to get this merged and into a point release to unblock some internal tooling improvements.

**Comment by 95-martin-orion** [comment]
Looks like this PR went out shortly after I left for a long weekend - reviewing now.

**Review by 95-martin-orion — APPROVED** [review]
This looks promising! There's no pressure for a qsim release at the moment, so we can iterate on this a bit (e.g. merge, attempt release, adjust if needed).

File: .github/workflows/release_wheels.yml [hunk 1]
```diff
@@ -39,7 +39,9 @@ jobs:
       CIBW_BEFORE_BUILD_MACOS: "brew install libomp"
       CIBW_REPAIR_WHEEL_COMMAND_MACOS: ""
       # due to package and module name conflict have to temporarily move it away to run tests
-      CIBW_BEFORE_TEST: "mv {package}/qsimcirq /tmp"
+      CIBW_BEFORE_TEST: >
+        mv {package}/qsimcirq /tmp
+        pip install -r {package}/dev-requirements.txt
```
[the remark this answers is not in the document]
Comment (95-martin-orion) [reply .github/workflows/release_wheels.yml]: Disregard this comment - I wrote it before a force push and couldn't delete it :/

**Comment by 95-martin-orion** [comment]
Release pipeline seems happy with this one. Thanks @lovesegfault for the fix!

**Pull request merged**